Sources/StreamVideoSwiftUI/Utils/PiPTrackSelectionUtils.swift

```diff
@@ -27,14 +27,16 @@ final class PiPTrackSelectionUtils {
             ) {
             return view
         }
-        
-        let firstId = participants.first?.id
-        if let firstId, let view = utils.videoRendererFactory.view(
-            for: firstId, isScreensharing: false
-        ) {
+
+        if
+            let firstId = participants.first?.id,
+            let view = utils.videoRendererFactory.view(for: firstId, isScreensharing: false)
+        {
             return view
+        } else {
+            log.warning("Unable to retrieve view for id:\(participants.first?.id ?? "n/a") with viewCount:\(utils.videoRendererFactory.views.count)")
         }
-        
-        return utils.videoRendererFactory.views.values.first
+
+        return nil
```

Comment:
Not sure if returning the first participant makes sense as it can be our own track